tests/conformance/test_conformance.py

```diff
@@ -252,6 +251,7 @@ def bucket_set_iam_policy(client, _preconditions, **resources):
     if _preconditions:
         bucket.set_iam_policy(policy)
     else:
+        policy.etag = None
```

Comment:
Maybe a comment here to explain why this is different from e.g. `bucket_update`? 
```suggestion
        # IAM policies have no metageneration:  clear ETag to avoid checking that it matches.
        policy.etag = None
```